### PR TITLE
feat: style delivery states

### LINKED ISSUE #14800: Style delivery states
We need to update delivery states:

- `sending` messages will be displayed using opacity, as per https://www.figma.com/file/PPWkgOYlZZDxZv5SDGsZVV/Posts-%26-Attachments-for-Mobile?node-id=3641%3A402567&t=F2uUp4Byflrd9uNA-4
- `sent` messages will be displayed normaly
- On a single press (not longpress), informations about `delivered` and `sent` will be displayed, and hidden after 3 seconds 

Out of scope: error state

### Designs

https://www.figma.com/file/PPWkgOYlZZDxZv5SDGsZVV/Posts-%26-Attachments-for-Mobile?node-id=3427%3A352700&t=SIJ83W5BaMTWRRSA-1

### PR DESCRIPTION
**Pull request by briansztamfater**
fixes #14800 

### Summary

This issue implements style to delivery states as per Figma designs. Delivery states are shown when user single-taps a message, and it is shown for just 3 seconds, then it disappears.

### Testing notes

- Sometimes it may be difficult to find and test all message outgoing states, I used waku1 fleet for testing but should work with waku2 as well
- The `sending` state, which makes the message to have opacity, may happen for a very very short time
- This PR does not include handling error states because it is out of scope of this issue
- As messages UI are built by different building blocks making the whole thing and there is pending work on other messages components, testing should be focused on just delivery states UI, take in mind that the whole messages design and some spacing may differ of how they should look when all elements are implemented as they should

#### Platforms

- Android
- iOS

##### Functional

- 1-1 chats
- public chats
- group chats

### Steps to test

- Open Status
- Create a chat or community with other users
- Send messages
- Verify delivery states UI 

status: ready

### CODE REVIEW

File: src/status_im2/contexts/chat/messages/content/view.cljs
Comment (ibrkhalil) [inline src/status_im2/contexts/chat/messages/content/view.cljs]: Is this change needed ?

File: src/status_im2/contexts/chat/messages/content/status/view.cljs
Comment (ibrkhalil) [inline src/status_im2/contexts/chat/messages/content/status/view.cljs]: Do we need to require these events ?

File: src/status_im2/contexts/chat/messages/content/status/style.cljs
Comment (ibrkhalil) [inline src/status_im2/contexts/chat/messages/content/status/style.cljs]: Extra line

**Review by ibrkhalil — APPROVED** [review]

File: src/status_im2/contexts/chat/messages/content/view.cljs
Comment (jo-mut) [reply src/status_im2/contexts/chat/messages/content/view.cljs]: if i recall well its the formatting we chose to go with using zprint

**Review by jo-mut — APPROVED** [review]

File: src/status_im2/contexts/chat/messages/content/status/view.cljs
Comment (ilmotta) [reply src/status_im2/contexts/chat/messages/content/status/view.cljs]: No, we really shouldn't. Events and subscriptions should be registered only once in the global re-frame registry. For events, that namespace is `status-im2.events`.

File: src/status_im2/contexts/chat/messages/content/view.cljs
Comment (ilmotta) [inline src/status_im2/contexts/chat/messages/content/view.cljs]: The number `3000` could be extracted into a var given its importance.

File: src/status_im2/contexts/chat/messages/content/view.cljs
Comment (ilmotta) [inline src/status_im2/contexts/chat/messages/content/view.cljs]: It always confuses me to see a reactive atom without a proper form-2 component. I've only seen it in the Status codebase. In React this kind of stuff is acceptable because `useState` is a hook with completely different capabilities, but in Reagent it's a different story.

There's probably a better way than resetting the state on every re-render of `message-with-reactions`. Even if it works @briansztamfater, this kind of practice can easily lead to bad performance due to the recreation of atoms on every render pass, and thus re-rendering of the entire sub-tree.

It's a dangerous practice. I also couldn't find a single example in Reagent docs and source examples doing this. Something tells me this is wrong.

File: src/status_im2/contexts/chat/messages/content/view.cljs
Comment (ilmotta) [reply src/status_im2/contexts/chat/messages/content/view.cljs]: `zprint` is configured to respect newlines. It might be the case that joining the lines will allow the map to be on the same line as `rn/view`.

Comment (ilmotta) [inline src/status_im2/contexts/chat/messages/content/view.cljs]: The style should be inside `{:style ...}`.

**Comment by VolodLytvynenko** [comment]
Hi @briansztamfater let me know when it is ready for tests. Thanx!

File: src/status_im2/contexts/chat/messages/content/status/view.cljs
Comment (briansztamfater) [reply src/status_im2/contexts/chat/messages/content/status/view.cljs]: Yes they shouldn't be here, will remove them

File: src/status_im2/contexts/chat/messages/content/view.cljs
Comment (briansztamfater) [reply src/status_im2/contexts/chat/messages/content/view.cljs]: Yes I don't know why there are some cases where style is set without `{:style ...}`, will replace the other cases in this file too

**Comment by briansztamfater** [comment]
> Hi @briansztamfater let me know when it is ready for tests. Thanx!

Sure, just pushed requested changes and rebased, once lord @ilmotta approves we can start testing

File: src/status_im2/contexts/chat/messages/content/view.cljs
Comment (ilmotta) [reply src/status_im2/contexts/chat/messages/content/view.cljs]: Thanks!

**Review by ilmotta — APPROVED** [review]
> > Hi @briansztamfater let me know when it is ready for tests. Thanx!
> 
> Sure, just pushed requested changes and rebased, once lord @ilmotta approves we can start testing

You got it @briansztamfater, take the approval!

Random image of the day:

image

**Comment by VolodLytvynenko** [comment]
HI @briansztamfater thank you for PR. Here is the list found issues

**ISSUE 1: The delivery status is shown using opacity** 

**Preconditions:**
USER_A and USER_B are mutual contacts

**Steps to reproduce:**
1. Open chat using USER_A and USER_B
2. Send the message as USER_A
3. Wait when it'll be delivered -> Tap on the message

**Actual result:**
The message is shown using opacity
https://user-images.githubusercontent.com/52490791/216326925-90817eb0-ae83-4926-bdb9-a6c5b132092d.mp4


**Expected result:**
The delivered message is shown normally
image


@briansztamfater I am not sure about expected result for ISSUE 2. Should the statuses be shown for edited messages after single tapping or 'edited' mark should be shown in this case? If the statuses should be shown then there is one more issue, currently 'edited' mark is not shown for messages after their updates
CC: @pedro-et take a look please

**ISSUE 2: The status is not correspond to required status  if the message is updated **

**Steps to reproduce:**
1. Open chat using USER_A and USER_B
2. Send the message as USER_A (now the status is 'delivered')
3. Close the app for USER_B
4. Long tap on the message -> Tap 'edit' -> update the message 
5. Single tap on the message

**Actual result:**
The status is still shown as 'delivered'

https://user-images.githubusercontent.com/52490791/216338751-908f3eae-c3b2-42eb-be6e-330b7ec01cf4.mp4


**Expected result:**
The status is changed to 'sent' status

**Comment by VolodLytvynenko** [comment]
@briansztamfater could you help with some questions please?
1. Should the sent images use the opacity? Currently they are not
2. How should the status of sent links and images be displayed considering that here is we have the functionality if the image or link is single tapped, then they should be opened?

**Comment by briansztamfater** [comment]
> HI @briansztamfater thank you for PR. Here is the list found issues
> 
> **ISSUE 1: The delivery status is shown using opacity**
> 
> **Preconditions:** USER_A and USER_B are mutual contacts
> 
> **Steps to reproduce:**
> 
> 1. Open chat using USER_A and USER_B
> 2. Send the message as USER_A
> 3. Wait when it'll be delivered -> Tap on the message
> 
> **Actual result:** The message is shown using opacity https://user-images.githubusercontent.com/52490791/216326925-90817eb0-ae83-4926-bdb9-a6c5b132092d.mp4
> 
> **Expected result:** The delivered message is shown normally image
> 
> @briansztamfater I am not sure about expected result for ISSUE 2. Should the statuses be shown for edited messages after single tapping or 'edited' mark should be shown in this case? If the statuses should be shown then there is one more issue, currently 'edited' mark is not shown for messages after their updates
> 
> **ISSUE 2: The status is not correspond to required status if the message is updated **
> 
> **Steps to reproduce:**
> 
> 1. Open chat using USER_A and USER_B
> 2. Send the message as USER_A (now the status is 'delivered')
> 3. Close the app for USER_B
> 4. Long tap on the message -> Tap 'edit' -> update the message
> 5. Single tap on the message
> 
> **Actual result:** The status is still shown as 'delivered'
> 
>  delivered.mp4 
> **Expected result:** The status is changed to 'sent' status

Thanks @VladimrLitvinenko for testing!

About issue 1, I was able to find the issue and will push a fix very soon.

For issue 2, I think we don't have clear specs about that. My thoughts are that we should keep the delivery state as it is and add an `Edited` label somewhere, but I think we should ask someone from the design team.

> @briansztamfater could you help with some questions please?
> 
> 1. Should the sent images use the opacity? Currently they are not
> 2. How should the status of sent links and images be displayed considering that here is we have the functionality if the image or link is single tapped, then they should be opened?

About these questions:

1. Images should use the opacity, for some reason they are not respecting the style. I am looking at it right now.
2. In my opinion user could tap outside to view the delivery state, but we should verify with the design team too

Either way, I'll check both cases and be right back soon

**Comment by VolodLytvynenko** [comment]
@briansztamfater directed the questions to the design team

**Comment by VolodLytvynenko** [comment]
@briansztamfater Pedro just replied in Discord

**My questions to Pedro**
Corresponding to design the message status is shown after single tap on it. https://www.figma.com/file/PPWkgOYlZZDxZv5SDGsZVV/Posts-%26-Attachments-for-Mobile?node-id=3641%3A402558&t=W564vX2DkQHXAtYK-0
Questions:
1. Should those statuses be shown for edited messages after single tapping or 'edited' mark should be shown in this case? If the statuses should be shown then as I understand the 'edited' mark should still be shown without tapping on the messages?
2. How should the status of sent links and images be displayed considering that here is we have the functionality if the image or link is single tapped, then they should be opened?

**Pedro replies**
1 yes edited should always show! 
2 clicking around the link within the box of the message

@briansztamfater add an 'edited' mark, please for edited messages. Currently, this mark  disappeared

**Comment by briansztamfater** [comment]
@VladimrLitvinenko thanks for bringing this information, I will add the Edited mark as well as the other fixes and will let you know

**Comment by briansztamfater** [comment]
@VladimrLitvinenko So, I talked with Pedro because I wasn't sure about the correct implementation of the Edited mark, and both delivery states and edited mark are different things. I think it should be covered on a different issue as it is out of scope of this one, but I found it was not too much extra effort to add it on this one so added the Edited mark into this one.

Attaching relevant conversation:
> pedro.et — Today at 3:43 PM
> https://www.figma.com/file/PPWkgOYlZZDxZv5SDGsZVV/Posts-%26-Attachments-for-Mobile?node-id=3277%3A336019&t=hTfMetZ8wHlAF7Qh-1
> sent and delivered states are always bellow with icon
> edited is text in a adifferent format in the same line if it fits
> 
> Brian Sztamfater — Today at 3:45 PM
> I see, so 'edited' mark and delivery states are different things
> 'edited' should be permanently shown right?
> and delivery states are shown on a single tap with the icon below, and just for 3 seconds
> 
> pedro.et — Today at 3:46 PM
> yes

For the rest of the issues I pushed the fixes too. Let me know how testing goes! :)

**Comment by VolodLytvynenko** [comment]
Hi @briansztamfater thanks for your work. The PR is tested and ready to be merged. Could you merge it, please?

**Comment by VolodLytvynenko** [comment]
Hi @briansztamfater I missed one detail. Some elements are not shown on the overlay after a long tap on the message with the 'sending' status.

**'Sending' status message after current PR**
Missed elements
- reply
- emojis 
image

**'Sending' status message before current PR**
- reply is shown
- emojis are shown
image

Could you help, Is it the expected result that those elements are not shown?

**Comment by briansztamfater** [comment]
Hi @VladimrLitvinenko does it make sense to react or reply to a message that was not sent yet? I can fix it quicky if necessary.

**Comment by VolodLytvynenko** [comment]
> Hi @VladimrLitvinenko does it make sense to react or reply to a message that was not sent yet? I can fix it quicky if necessary.

@briansztamfater Agree with you. I asked the design team and will let you know when they provide me with any news. Thanx!

**Pull request merged**
